File: xl-job-ui/src/api/joblog.ts
import request, { ApiResponse, PageResponse } from './request'

// 日志相关类型
export interface JobLog {
  id: number
  jobGroup: number
  jobId: number
  executorAddress: string
  executorHandler: string
  executorParam: string
  executorShardingParam: string
  executorFailRetryCount: number
  triggerTime: number
  triggerCode: number
  triggerMsg: string
  handleTime: number
  handleCode: number
  handleMsg: string
  alarmStatus: number
  jobGroupName?: string
}

export interface JobLogQuery {
  jobGroup?: number
  jobId?: number
  logId?: number
  filterTime?: string
  triggerTimeStart?: string
  triggerTimeEnd?: string
  start?: number
  length?: number
}

export interface LogDetail {
  executorAddress: string
  triggerCode: number
  triggerMsg: string
  handleCode: number
  handleMsg: string
}

// 获取日志分页列表
export function getJobLogPage(params: JobLogQuery): Promise<ApiResponse<PageResponse<JobLog>>> {
  return request.get('/joblog/pageList', { params })
}

// 获取日志详情
export function getJobLogDetail(id: number): Promise<ApiResponse<LogDetail>> {
  return request.get('/joblog/logDetailPage', { params: { id } })
}

// 获取执行日志
export function getExecutionLog(executorAddress: string, logId: number, fromLineNum: number): Promise<ApiResponse<{
  content: string
  end: boolean
}>> {
  return request.get('/joblog/logDetailCat', {
    params: { executorAddress, logId, fromLineNum }
  })
}

// 终止任务
export function killJob(logId: number): Promise<ApiResponse<string>> {
  return request.post('/joblog/logKill', { logId })
}

// 清除日志
export function clearLog(jobGroup: number, jobId: number, type: number): Promise<ApiResponse<string>> {
  return request.post('/joblog/clearLog', { jobGroup, jobId, type })
}

File: xl-job-admin/src/main/java/com/xl/job/admin/entity/JobLog.java
package com.xl.job.admin.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 执行日志实体
 */
@Data
@TableName("xl_job_log")
public class JobLog {

    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 执行器组名
     */
    private String jobGroup;

    /**
     * 任务ID
     */
    private Long jobId;

    /**
     * 执行器地址
     */
    private String executorAddress;

    /**
     * 任务Handler
     */
    private String executorHandler;

    /**
     * 任务参数
     */
    private String executorParam;

    /**
     * 分片参数
     */
    private String executorShardingParam;

    /**
     * 失败重试次数
     */
    private Integer executorFailRetryCount;

    /**
     * 调度时间
     */
    private LocalDateTime triggerTime;

    /**
     * 调度结果码
     */
    private Integer triggerCode;

    /**
     * 调度日志
     */
    private String triggerMsg;

    /**
     * 执行时间
     */
    private LocalDateTime handleTime;

    /**
     * 执行结果码
     */
    private Integer handleCode;

    /**
     * 执行日志
     */
    private String handleMsg;

    /**
     * 告警状态:0-默认,1-无需告警,2-已告警
     */
    private Integer alarmStatus;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;
}
